third_party/move/tools/move-coverage/src/source_coverage.rs

```diff
@@ -236,18 +367,20 @@ impl SourceCoverage {
             }
             writeln!(output_writer)?;
         }
+        color.undo();
         Ok(())
     }
 }
 
-fn merge_spans(cov: FunctionSourceCoverage) -> Vec<Span> {
+fn merge_spans(file_hash: FileHash, cov: FunctionSourceCoverage) -> Vec<Span> {
     if cov.uncovered_locations.is_empty() {
         return vec![];
     }
 
     let mut covs: Vec<_> = cov
         .uncovered_locations
         .iter()
+        .filter(|loc| loc.file_hash() == file_hash)
```

Comment:
This change introduces a new crash: https://github.com/aptos-labs/aptos-core/issues/14454

It can be fixed by adding a check that `if `covs.is_empty()` { return vec![]; }` right after creating `covs` here.

Reply:
I had filed the crash while reviewing, but I did not realize until after filing the bug that the root cause was this change.

Reply:
Done.

Reply:
If you have tested the scenario mentioned in the issue and it is fixed, then you can mention in the PR description that it also fixes that issue so that it gets closed.